presto-main/src/main/java/com/facebook/presto/execution/QueryLimitMemoryRevokingScheduler.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.memory.LocalMemoryManager;
+import com.facebook.presto.memory.MemoryPool;
+import com.facebook.presto.memory.MemoryPoolListener;
+import com.facebook.presto.memory.QueryContext;
+import com.facebook.presto.memory.VoidTraversingQueryContextVisitor;
+import com.facebook.presto.operator.OperatorContext;
+import com.facebook.presto.operator.TaskContext;
+import com.facebook.presto.spi.QueryId;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import static com.facebook.presto.execution.MemoryRevokingUtils.getMemoryPools;
+import static java.util.Objects.requireNonNull;
+
+public class QueryLimitMemoryRevokingScheduler
+{
+    private static final Logger log = Logger.get(QueryLimitMemoryRevokingScheduler.class);
+
+    private final ScheduledExecutorService taskManagementExecutor;
+    private final Function<QueryId, QueryContext> queryContextSupplier;
+
+    // Technically not thread safe but should be fine since we only call this on PostConstruct and PreDestroy.
+    // PreDestroy isn't called until server shuts down/ in between tests.
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
```

Comment:
I might be missing something, but this field does not appear to ever be assigned to.

Reply:
ah yes, thanks. I copied this class from the TaskThresholdMemoryRevokingScheduler but got rid of the regular asynchronous checks since they didn't seem useful.